src/share/classes/com/sun/javatest/regtest/config/RegressionEnvironment.java

```diff
@@ -46,5 +48,27 @@ public TestEnvironment copy() {
         return new RegressionEnvironment(this);
     }
 
+    /**
+     * {@return the hostname of the host in which this test harness is being run}
+     */
+    public final String getHostName() {
+        return LazyHostNameLookup.HOST_NAME;
+    }
+
     public final RegressionParameters params;
+
+
+    private static final class LazyHostNameLookup {
```

Comment:
LazyHostNameLookup infers a lookup deferred to the latest possible moment While 
CachedHostName would convey a more appropriate semantics.

The rational for the change is the cost of a getCanonicalHostName invocation. However, InetAddress caches a getLocalHost
call in a CachedLocalHost, which proves a mapping between a hostname and an InetAddress, but this cached value is not used in the getCanonicalHostName invocation. This seems strange that the cached values in InetAddress are not used more effectively for return a Hostname !!

Reply:
@msheppar Is this worth a P4 followup RFE against InetAddress?

Reply:
Hello Mark,

> LazyHostNameLookup infers a lookup deferred to the latest possible moment While
CachedHostName would convey a more appropriate semantics.

I have now updated the PR to use the term "Cached" instead of "Lazy". No other changes. Tests continue to pass.

Reply:
cheers JP, thanks for considering the suggestion